zerver/tests/test_signup.py

```diff
@@ -2838,6 +2838,33 @@ def test_create_two_realms(self) -> None:
         # The remaining PreregistrationUser should have been used up:
         self.assertEqual(PreregistrationUser.objects.filter(email=email, status=0).count(), 0)
 
+    def test_set_default_lanugage_using_browser_locale_during_realm_creation(self) -> None:
+        password = "test"
+        string_id = "zuliptest"
+        email = "user1@test.com"
+        realm_name = "Test"
+
+        result = self.client_post("/new/", {"email": email})
+        self.assertEqual(result.status_code, 302)
+        self.assertTrue(result["Location"].endswith(f"/accounts/new/send_confirm/{email}"))
+        confirmation_url = self.get_confirmation_url_from_outbox(email)
+        result = self.client_get(confirmation_url)
+        self.assertEqual(result.status_code, 200)
+
+        result = self.submit_reg_form_for_user(
+            email,
+            password,
+            realm_subdomain=string_id,
+            realm_name=realm_name,
+            HTTP_ACCEPT_LANGUAGE="fr,en,hi",
+        )
+
+        realm = get_realm(string_id)
+        user = get_user_by_delivery_email(email, realm)
+
+        self.assertEqual(realm.default_language, "fr")
+        self.assertEqual(user.default_language, "fr")
+
```

Comment:
Hmm. Tests should not be in a separate commit. Keep it alongside the associated commit. Is there a specific reason why you choose to do it like this?

Reply:
I just wanted to show the process how I created flow of the PR. so, I thought to create a separate commit for the tests. But, now I have merged them in a single commit.